lnclient/lnd/lnd.go

```diff
@@ -562,15 +564,35 @@ func (svc *LNDService) Shutdown() error {
 }
 
 func (svc *LNDService) GetNodeConnectionInfo(ctx context.Context) (nodeConnectionInfo *lnclient.NodeConnectionInfo, err error) {
-	info, err := svc.client.GetInfo(ctx, &lnrpc.GetInfoRequest{})
+	pubkey := svc.GetPubkey()
+	nodeInfo, err := svc.client.GetNodeInfo(ctx, &lnrpc.NodeInfoRequest{
+		PubKey: pubkey,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	addresses := nodeInfo.Node.Addresses
+
+	if addresses == nil || len(addresses) < 1 {
+		return nil, errors.New("no available listening addresses")
```

Comment:
we should not return an error here, otherwise the pubkey is also not returned

Reply:
Fixed to always return pubkey